### PR TITLE
OnConnect.

### PR DESCRIPTION
**Pull request by brian-js**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/738)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/routing_node.rs [hunk 1]
```diff
@@ -189,7 +189,8 @@ impl RoutingNode {
                     // the concept of a first node, as it is just hidden, not logically removed
                     // refactoring to crust 0.3 has made this logic even worse than it was.
                     if self.core.is_node() {
```
Comment (benjaminbollen) [inline src/routing_node.rs]: can we get rid of this self.core.is_node() altogether; directly go to handle_on_connect() and inside that function we can send a (client_)hello (if we are a client)

File: src/routing_node.rs [hunk 2]
```diff
@@ -232,6 +233,33 @@ impl RoutingNode {
...
 
+    fn handle_on_connect(&mut self, connection: ::crust::Connection) {
+        match self.core.state() {
+            &::routing_core::State::Disconnected => {
+                self.core.add_bootstrap_connection(connection.clone());
+                ignore(self.send_hello(connection, None));
+                return;
+            },
+            &::routing_core::State::Bootstrapped => {
+                self.crust_service.drop_node(connection);
```
Comment (benjaminbollen) [inline src/routing_node.rs]: combined with above comment, here we would also send a hello (which will indicate that we are a client connecting)

File: src/routing_node.rs [hunk 3]
```diff
@@ -232,6 +233,33 @@ impl RoutingNode {
           self.set_cache_options(preserve_cache_options);
     }
 
+    fn handle_on_connect(&mut self, connection: ::crust::Connection) {
+        match self.core.state() {
+            &::routing_core::State::Disconnected => {
```
Comment (benjaminbollen) [inline src/routing_node.rs]: can you add small inline comments on the motivation behind these steps?  that would be very valuable here I think.  Thanks!

**Comment by benjaminbollen** [comment]
looks really good thx

**Comment by benjaminbollen** [comment]
src/routing_node.rs:287:5: 306:6 error: method is never used: `handle_new_bootstrap_connection` ;)

**Pull request merged**